libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,511 @@
+  std::advance(it, start_node);
+  GNode source = *it;
+  it = graph.begin();
+  std::advance(it, report_node);
+  GNode report = *it;
+
+  size_t approxNodeData = graph.size() * 64;
+  katana::Prealloc(1, approxNodeData);
+  katana::ReportPageAllocGuard page_alloc;
+
+  katana::do_all(katana::iterate(graph), [&](const GNode& n) {
+    graph.template GetData<NodeMax<Weight>>(n) = 0;
+    graph.template GetData<NodeCount<Weight>>(n) = 0;
+  });
+
+  katana::StatTimer execTime("kSSSP");
+  execTime.start();
+
+  katana::InsertBag<std::pair<Weight, Path*>> paths;
+  katana::InsertBag<Path*> path_pointers;
+
+  bool reachable = true;
+
+  switch (algo_reachability) {
+  case async:
+    reachable = CheckReachabilityAsync<GraphTy, Weight, BFSUpdateRequest>(
+        &graph, source, BFSReqPushWrap(), BFSOutEdgeRangeFn{&graph},
+        report_node);
+    break;
+  case syncLevel:
+    reachable = CheckReachabilitySync<GraphTy, Weight>(&graph, source, report_node);
+    break;
+  default:
+    std::abort();
+  }
+
+  PathAlloc path_alloc;
+
+  if (reachable) {
+    switch (plan.algorithm()) {
+    case kSsspPlan::kDeltaTile:
+      DeltaStepAlgo<GraphTy, Weight, kSSSPSrcEdgeTile, OBIM>(
+          &graph, source, kSSSPSrcEdgeTilePushWrap{&graph}, kSSSPTileRangeFn(),
+          &paths, &path_pointers, path_alloc, report_node, num_paths,
+          step_shift);
+      break;
+    case kSsspPlan::kDeltaStep:
+      DeltaStepAlgo<GraphTy, Weight, kSSSPUpdateRequest, OBIM>(
+          &graph, source, kSSSPReqPushWrap(), kSSSPOutEdgeRangeFn{&graph},
+          &paths, &path_pointers, path_alloc, report_node, num_paths,
+          step_shift);
+      break;
+    case kSsspPlan::kDeltaStepBarrier:
+      katana::gInfo("Using OBIM with barrier\n");
+      DeltaStepAlgo<GraphTy, Weight, kSSSPUpdateRequest, OBIM_Barrier>(
+          &graph, source, kSSSPReqPushWrap(), kSSSPOutEdgeRangeFn{&graph},
+          &paths, &path_pointers, path_alloc, report_node, num_paths,
+          step_shift);
+      break;
+
+    default:
+      return katana::ErrorCode::InvalidArgument;
+    }
+  }
+
+  execTime.stop();
+  page_alloc.Report();
+
+  if (reachable) {
+    std::multimap<uint32_t, Path*> paths_map;
+
+    for (auto pair : paths) {
+      paths_map.insert(std::make_pair(pair.first, pair.second));
+    }
+
+    katana::gPrint("Node ", report, " has these k paths:\n");
+
+    uint32_t num =
+        (paths_map.size() > num_paths) ? num_paths : paths_map.size();
+
+    auto it_report = paths_map.begin();
+
+    for (uint32_t iter = 0; iter < num; iter++) {
+      const Path* path = it_report->second;
+      PrintPath(path);
+      katana::gPrint(" ", report, "\n");
+      katana::gPrint("Weight: ", it_report->first, "\n");
+      it_report++;
+    }
+
+    katana::do_all(katana::iterate(path_pointers), [&](Path* p) {
+      path_alloc.DeletePath(p);
+    });
+  }
+
+  return katana::ResultSuccess();
+}
+
+template <typename GraphTy>
+katana::Result<void>
+KssspSetWeight(
```

Comment:
just rename this to `Ksssp`. Since the other `Ksssp` function has different set of arguments, this should work.